contrib/build-meep.sh

```diff
@@ -63,6 +72,28 @@ Press return to continue
 EOF
 read junk
 
+if $ubuntu; then
+    RPATH_FLAGS="-Wl,-rpath,${DESTDIR}/lib:/usr/lib/x86_64-linux-gnu/hdf5/openmpi"
+    LDFLAGS="-L${DESTDIR}/lib -L/usr/lib/x86_64-linux-gnu/hdf5/openmpi ${RPATH_FLAGS}"
+    CFLAGS="-I${DESTDIR}/include -I/usr/include/hdf5/openmpi"
+fi
+
+if $centos; then
+    export CC=/usr/lib64/openmpi/bin/mpicc
+    export CXX=/usr/lib64/openmpi/bin/mpicxx
+    export PATH=${PATH}:/usr/lib64/openmpi/bin
+    RPATH_FLAGS="-Wl,-rpath,${DESTDIR}/lib64:/usr/lib64/openmpi/lib"
+    LDFLAGS="-L${DESTDIR}/lib64 -L/usr/lib64/openmpi/lib ${RPATH_FLAGS}"
+    CFLAGS="-I${DESTDIR}/include -I/usr/include/openmpi-x86_64/"
+    export PYTHONPATH=${DESTDIR}/lib/python3.6/site-packages
+    export PYTHONPATH=${PYTHONPATH}:/usr/local/lib64/python3.6/site-packages
+    export PYTHONPATH=${PYTHONPATH}:/usr/local/lib64/python3.6/site-packages/mpi4py
+    export LD_LIBRARY_PATH=${DESTDIR}/lib64:/usr/lib64/openmpi/lib:/usr/local/lib64/python3.6/site-packages/mpi4py/lib-pmpi:/usr/local/lib64/python3.6/site-packages/mpi4py
+    export LD_PRELOAD=/usr/lib64/openmpi/lib/libmpi.so
```

Comment:
Why is preloading needed?

Reply:
> Why is preloading needed?

I don't know the reason TBH but I can show the effect of not using it with centos 7.

If I'm missing something I don't know yet where. This is not needed for ubuntu.

```
[root@771c413e32e4 /]# python3 /tmp/test-meep.py 

**
** failed to load python MPI module (mpi4py)
** /usr/local/lib64/python3.6/site-packages/mpi4py/MPI.cpython-36m-x86_64-linux-gnu.so: undefined symbol: ompi_mpi_logical8 
**

Vector3<16.0, 8.0, 0.0>

Elapsed run time = 0.0002 s
[root@771c413e32e4 /]# export LD_PRELOAD=/usr/lib64/openmpi/lib/libmpi.so
[root@771c413e32e4 /]# python3 /tmp/test-meep.py 
Using MPI version 3.0, 1 processes
Vector3<16.0, 8.0, 0.0>

Elapsed run time = 0.0002 s
[root@771c413e32e4 /]# 
```
(It is a docker build <= me trying to explain 'root' :)